Src/IronPython.Modules/nt.cs

```diff
@@ -351,9 +351,25 @@ public static int dup(CodeContext/*!*/ context, int fd) {
             PythonFileManager fileManager = context.LanguageContext.FileManager;
 
             StreamBox streams = fileManager.GetStreams(fd); // OSError if fd not valid
-            fileManager.EnsureRefStreams(streams);
-            fileManager.AddRefStreams(streams);
-            return fileManager.Add(new(streams));
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+                fileManager.EnsureRefStreams(streams);
+                fileManager.AddRefStreams(streams);
+                return fileManager.Add(new(streams));
+            } else {
+                if (!streams.IsSingleStream && fd is 1 or 2) {
+                    // If there is a separate write stream, dupping over stout or sderr uses write stream's file descriptor
+                    fd = streams.WriteStream is FileStream fs ? fs.SafeFileHandle.DangerousGetHandle().ToInt32() : fd;
```

Comment:
Is it documented somewhere that this is the file descriptor? Or would we be relying on an implementation detail? Not that I'm against it, just curious...

Reply:
This is an implementation detail. The documentation only reads that it is "platform-specific", as far as I could find.

Reply:
BTW, this piece of code is a hack around the dual-stream case (used for mode `ab+`), which I intend to abolish, in probably the next PR, so this piece will be deleted as well.